### PR TITLE
Refactor Qwen3 14B prefill fwd

### PR DESCRIPTION
**Pull request by ndleslx**
## Summary
- Rename the Qwen3 14B prefill entrypoint to prefill_fwd.py and split the layer body into jit/jit.inline form.
- Add full-layer prefill execution with local intermediate tensors and dynamic input token count binding.
- Gather the final prefill token into a static [BATCH, HIDDEN] tensor and run RMSNorm + LM head to produce logits.

## Validation
- python -m py_compile models/qwen3/14b/prefill_fwd.py models/qwen3/14b/decode_fwd.py models/qwen3/14b/decode_layer.py models/qwen3/14b/rms_lm_head.py
- PTO2_RING_HEAP=536870912 PTO2_RING_TASK_WINDOW=131072 PTO2_RING_DEP_POOL=131072 task-submit --device auto --run "python models/qwen3/14b/prefill_fwd.py -p a2a3 --batch 1 --num-layers 2"

### CODE REVIEW

**Comment by ndleslx** [comment]
@CodeRabbit pls re-review this pull request
